funderfinder/sources/_finder.py

```diff
@@ -0,0 +1,55 @@
+from typing import Union
+
+
+class Finder:
+    name = "Abstract Funder Finder"
+
+    @staticmethod
+    def get_repo_name(project: str) -> str:
+        """
+        Extracts a repo name from a github identifier (i.e. funder-finder from georgetown-cset/funder-finder)
+        :param project: Github project name
+        :return: Repo name
+        """
+        return project.strip().split("/")[1]
+
+    @staticmethod
+    def get_owner_name(project: str) -> str:
+        """
+        Extracts an owner (user or organization) name from a github identifier (i.e. georgetown-cset
+        from georgetown-cset/funder-finder)
+        :param project: Github project name
+        :return: Owner name
+        """
+        return project.strip().split("/")[0]
+
+    def run(
+        self,
+        gh_project_slug: Union[str, None] = None,
+        project_name: Union[str, None] = None,
+    ) -> Union[dict, None]:
+        """
+        This method should be implemented for subclasses and, if funding is found for a given source, return a
+        nonempty subset of the following information
+        {
```

Comment:
Great use of the ontology we developed! Having it in the comment is very helpful.